localstack/services/lambda_/invocation/executor_endpoint.py

```diff
@@ -36,97 +37,128 @@ def __init__(self, message):
         super().__init__(message)
 
 
-class ExecutorEndpoint:
+class Endpoint(abc.ABC):
+    @abc.abstractmethod
+    def invocation_response(self, request: Request, req_id: str) -> Response:
+        pass
+
+    @abc.abstractmethod
+    def invocation_error(self, request: Request, req_id: str) -> Response:
+        pass
+
+    @abc.abstractmethod
+    def invocation_logs(self, request: Request, invoke_id: str) -> Response:
+        pass
+
+    @abc.abstractmethod
+    def status_ready(self, request: Request, executor_id: str) -> Response:
+        pass
+
+    @abc.abstractmethod
+    def status_error(self, request: Request, executor_id: str) -> Response:
+        pass
+
+
+class ExecutorRouter:
+    endpoints: dict[str, Endpoint]
+
+    def __init__(self):
+        self.endpoints = {}
+
+    def register_endpoint(self, executor_id: str, endpoint: Endpoint):
+        self.endpoints[executor_id] = endpoint
+
+    def unregister_endpoint(self, executor_id: str):
+        self.endpoints.pop(executor_id)
+
+    @route(f"{NAMESPACE}/<executor_id>/invocations/<req_id>/response", methods=["POST"])
+    def invocation_response(self, request: Request, executor_id: str, req_id: str) -> Response:
+        endpoint = self.endpoints[executor_id]
+        return endpoint.invocation_response(request, req_id)
+
+    @route(f"{NAMESPACE}/<executor_id>/invocations/<req_id>/error", methods=["POST"])
+    def invocation_error(self, request: Request, executor_id: str, req_id: str) -> Response:
+        endpoint = self.endpoints[executor_id]
+        return endpoint.invocation_error(request, req_id)
+
+    @route(f"{NAMESPACE}/<executor_id>/invocations/<invoke_id>/logs", methods=["POST"])
+    def invocation_logs(self, request: Request, executor_id: str, invoke_id: str) -> Response:
+        endpoint = self.endpoints[executor_id]
+        return endpoint.invocation_logs(request, invoke_id)
+
+    @route(f"{NAMESPACE}/<env_id>/status/<executor_id>/ready", methods=["POST"])
+    def status_ready(self, request: Request, env_id: str, executor_id: str) -> Response:
+        endpoint = self.endpoints[executor_id]
+        return endpoint.status_ready(request, executor_id)
+
+    @route(f"{NAMESPACE}/<env_id>/status/<executor_id>/error", methods=["POST"])
+    def status_error(self, request: Request, env_id: str, executor_id: str) -> Response:
+        endpoint = self.endpoints[executor_id]
+        return endpoint.status_error(request, executor_id)
+
+
+@singleton_factory
+def executor_router():
+    router = ExecutorRouter()
+    ROUTER.add(router)
+    return router
```

Comment:
not the best solution to have this global, but :shrug: probably better than to add extra layers right now